mllib/src/main/scala/org/apache/spark/ml/util/ReadWrite.scala

```diff
@@ -471,3 +471,26 @@ private[ml] object MetaAlgorithmReadWrite {
     List((instance.uid, instance)) ++ subStageMaps
   }
 }
+
+class FileSystemOverwrite extends BaseReadWrite with Logging {
```

Comment:
Also, I'd make this a static object, eliminate the BaseReadWrite dependency, and just take ```sc``` as an argument.